src/amo/components/VPNPromoBanner/index.js

```diff
@@ -135,7 +155,7 @@ export class VPNPromoBannerBase extends React.Component<InternalProps> {
   render(): null | React.Node {
     const { i18n } = this.props;
 
-    if (!this.shouldShowBanner()) {
+    if (!this.shouldShowBanner() || !this.state.showPromo) {
```

Comment:
I'm not sure I understand the question, or rather why it is being asked. One case where that could happen is if a user has dismissed the banner, and they are also landing on an `/android/` page. In that case the check for `this.shouldShowBanner()` would tell us not to show it, and we wouldn't need the check for `this.state.showPromo`, but if the same user was to land on a `/firefox/` page then we'd need the check for `this.state.showPromo` in order to hide the banner. But maybe I am missing the point of this question.

Reply:
Ok, maybe I understand your point after reading Will's suggestion below. Is it simply that I could make the check for `this.state.showPromo` part of `this.shoudShowBanner`?

Reply:
I just thought the same conditions that would lead to `this.state.showPromo` being set to false would also lead to `this.showShowBanner()` being false

`this.state = { showPromo: props.variant === VARIANT_SHOW }` but `shouldShowBanner` does `return variant === VARIANT_SHOW &&   ...`

Reply:
That's true, and I've added a commit to the patch to deal with that, but it's not the only way that `this.state.showPromo` can change. The reason this exists in the first place is to hide the banner when a user clicks on it, at which point the value changes from `true` to `false`.